src/RequestManager.ts

```diff
@@ -103,20 +104,25 @@ class RequestManager {
 
     const batch = JSON.stringify(this.batch);
     this.batch = [];
-    this.transports[0].sendData(batch);
+    this.getTransport().sendData(batch);
   }
 
-  private onData(data: string): void {
+  public onError(callback: (error: Error) => void): void {
+    const transport = this.getTransport();
+    transport.onError(callback);
+  }
+
+  private onData(data: string, onError: (error: Error) => void): void {
```

Comment:
nit. onError name might be confusing as an argument name here, feel free to ignore